packages/@glimmer/runtime/lib/vm/element-builder.ts

```diff
@@ -84,7 +84,7 @@ export class NewElementBuilder implements ElementBuilder {
   [CURSOR_STACK] = new Stack<Cursor>();
   private modifierStack = new Stack<Nullable<ModifierInstance[]>>();
   private blockStack = new Stack<LiveBlock>();
-  private htmlElementsState: { shouldAddHtmlElement?: boolean }[];
+  private htmlElementsState = new Set();
```

Comment:
Maybe you are not fully done, but to be clear about what I meant in the comment – in the final state of the PR (w.r.t. "I think it would be fine if we just want to add modifiers to the tree without adding elements"), this should not even be needed, and that we wouldn't have added the new "html-element" type to the tree at all. When I say you can infer and group them in the inspector if you want to, I mean you can do that using the bounds, which would presumably reference the element they are attached to, and then you can recreate the grouping with a weak map or something on the inspector side.

And, yes, if it's feasible and about the same amount of code, it would probably be a good idea to do the work in `pushModifier` rather than in `closeElement`, all else being equal.

Reply:
so, now there is a html element which wraps only the modifiers. the rest can be done in inspector

Reply:
As I was saying above, let’s not even add that

Reply:
updated